src/Designer/frontend/libs/studio-browser-storage/src/CookieStorage/CookieStorage.test.ts

```diff
@@ -0,0 +1,344 @@
+      expect(result).toBe(specialValue);
+    });
+
+    it('should not store undefined values', () => {
+      const consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation();
+      CookieStorage.setItem('undefinedKey', undefined);
+      const result = CookieStorage.getItem<string>('undefinedKey');
+      expect(result).toBeNull();
+      expect(consoleWarnSpy).toHaveBeenCalledWith(
+        expect.stringContaining('Cannot set cookie "undefinedKey"'),
+      );
+      consoleWarnSpy.mockRestore();
+    });
+
+    it('should not store null values', () => {
+      const consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation();
+      CookieStorage.setItem('nullKey', null);
+      const result = CookieStorage.getItem<string>('nullKey');
+      expect(result).toBeNull();
+      expect(consoleWarnSpy).toHaveBeenCalledWith(
+        expect.stringContaining('Cannot set cookie "nullKey"'),
+      );
+      consoleWarnSpy.mockRestore();
+    });
+
+    it('should overwrite existing cookie with same key', () => {
+      CookieStorage.setItem('overwriteKey', 'firstValue');
+      CookieStorage.setItem('overwriteKey', 'secondValue');
+      const result = CookieStorage.getItem<string>('overwriteKey');
+      expect(result).toBe('secondValue');
+    });
+  });
+
+  describe('removeItem', () => {
+    it('should remove an existing cookie', () => {
+      CookieStorage.setItem('removeKey', 'value');
+      expect(CookieStorage.getItem<string>('removeKey')).toBe('value');
+
+      CookieStorage.removeItem('removeKey');
+      expect(CookieStorage.getItem<string>('removeKey')).toBeNull();
+    });
+
+    it('should not throw error when removing non-existent cookie', () => {
+      expect(() => CookieStorage.removeItem('nonExistentKey')).not.toThrow();
+    });
+
+    it('should remove cookie with matching path', () => {
+      CookieStorage.setItem('pathKey', 'value', { path: '/test' });
+      CookieStorage.removeItem('pathKey', { path: '/test' });
+      expect(CookieStorage.getItem<string>('pathKey')).toBeNull();
+    });
+  });
+
+  describe('getAllKeys', () => {
+    it('should return empty array when no cookies exist', () => {
+      const keys = CookieStorage.getAllKeys();
+      expect(keys).toEqual([]);
+    });
+
+    it('should return all cookie keys', () => {
+      CookieStorage.setItem('key1', 'value1');
+      CookieStorage.setItem('key2', 'value2');
+      CookieStorage.setItem('key3', 'value3');
+
+      const keys = CookieStorage.getAllKeys();
+      expect(keys).toHaveLength(3);
+      expect(keys).toContain('key1');
+      expect(keys).toContain('key2');
+      expect(keys).toContain('key3');
+    });
+
+    it('should decode URL-encoded keys', () => {
+      CookieStorage.setItem('key with spaces', 'value');
+      const keys = CookieStorage.getAllKeys();
+      expect(keys).toContain('key with spaces');
+    });
+  });
+
+  describe('cookie options', () => {
+    it('should set cookie with expiration date', () => {
```

Comment:
Småpirk: Det var litt vanskelig å forstå hva denne testen skulle teste. Savner informasjon om hvilken funksjon som testes og hva som er ventet oppførsel. Her er et forslag:
```suggestion
    test('getItem should return the value of a given cookie when it has been set with an expiration date in the future', () => {
```
Mye av det samme går egentlig igjen i noen av de andre testbeskrivelsene - de er litt mangelfulle. Jeg tenker minimum informasjon i en testbeskrivelse bør være dette:
- Hvilke(n) funksjon(er)/metode(r) er det som testes (OK når funksjonen står i `describe`)
- Hvilke kriterier (inndata eller eksterne data som funksjonen påvirkes av) testes den under
- Ventet resultat under gitte kriterier

Når det er sagt, forstår jeg at dette er en smakssak og at det bør være tilstrekkelig å lese selve koden i testen - det er jo bare der man alltid finner sannheten. Jest er jo også et av få testrammeverk som jeg kjenner til som i det hele tatt har denne muligheten til fritekstbeskrivelser; det later til å være en Javascript-greie.